src/mpid/ch4/netmod/ofi/ofi_send.h

```diff
@@ -30,7 +30,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_send_lightweight(const void *buf,
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_SEND_LIGHTWEIGHT);
     match_bits = MPIDI_OFI_init_sendtag(comm->context_id + context_offset, comm->rank, tag, 0);
     mpi_errno =
-        MPIDI_OFI_send_handler(MPIDI_Global.ctx[0].tx, buf, data_sz, NULL, comm->rank,
+        MPIDI_OFI_inject_handler(MPIDI_Global.ctx[0].tx, buf, data_sz, NULL, comm->rank,
                                MPIDI_OFI_av_to_phys(addr), match_bits,
                                NULL, MPIDI_OFI_DO_INJECT, MPIDI_OFI_CALL_LOCK);
```

Comment:
We can probably drop the `MPIDI_OFI_DO_INJECT` if we're going to have separate calls for send/inject.

On the other hand, why does this need to be a separate function? Why can't it be in `MPIDI_OFI_send_handler`?

Reply:
For that matter, why do we want to bubble `FI_EAGAIN` up to the user at all? If we're going to do that, I think we should do it after some limited number of retries since this is not an unexpected result from an OFI call. Can we just change the `MPIDI_OFI_CALL_RETRY` macro to do that?

Reply:
Limited number of retries is probably a good idea. This is really just an experiment. Task-based runtimes that use MPI for communication might be able to context switch to something else if they get this type of error.

Reply:
Please use a CVAR for the number of retries, so users/tools can control it.

Reply:
It's true that the send handler could also run into an eagain. The reason I left that as-is is because at that time, a send request has already been allocated. If we wanted to give up on that operation and return an error to the user, there's a little more cleanup to do first.

Reply:
You could put it only in the inject part of that function.